Added adminpanel functionality for employment rates

palbock opened the pull request.

Commit:
Added adminpanel functionality for employment rates

klyngen commented on packages/adminpanel/src/components/EmploymentRateTable.tsx:
```diff
@@ -0,0 +1,80 @@
+    id: number;
+  };
+  const { alvtimeFetcher } = useContext(AlvtimeContext);
+  const columns: Column<object>[] = [
```
Det er litt fyfy å bruke typer som `any` og `object`. Det er bedre å lage et interface
```
interface ColumnData {
  title: string;
  field: 'string';
  editable: 'string;
  type: 'string'
}
```

klyngen commented on packages/adminpanel/src/components/EmploymentRateTable.tsx:
```diff
@@ -0,0 +1,80 @@
+    setCache(path, [...addedData, ...data]);
+  };
+
+  const handleRowUpdate = async (newData: any, oldData: any) => {
```
Klarer du å legge til skikkelige typer her?
Bruk gjerne http://json2ts.com/ til å lage interfaces fra json

palbock commented:
Jeg trenger litt hjelp til å komme helt i mål. Har forsøkt å lage klasser her nå men får denne feilmeldingen:
@klyngen @Trulsaa 
```
ERROR in src/components/EmploymentRateTable.tsx:86:5
TS2769: No overload matches this call.
  Overload 2 of 2, '(props: MaterialTableProps<ColumnData>, context: any): MaterialTable<ColumnData>', gave the following error.
    Type '{ actionsCellStyle?: React.CSSProperties | undefined; detailPanelColumnStyle?: React.CSSProperties | undefined; editCellStyle?: React.CSSProperties | undefined; ... 57 more ...; cspNonce?: string | undefined; }' is not assignable to type 'Options<ColumnData>'.
      Types of property 'exportFileName' are incompatible.
        Type 'string | ((columns: Column<object>, data: string[][]) => string) | undefined' is not assignable to type 'string | ((columns: Column<ColumnData>, data: string[][]) => string) | undefined'.
  Overload 2 of 2, '(props: MaterialTableProps<ColumnData>, context: any): MaterialTable<ColumnData>', gave the following error.
    Type '(newData: CreateRateData) => Promise<void>' is not assignable to type '(newData: ColumnData) => Promise<any>'.
  Overload 2 of 2, '(props: MaterialTableProps<ColumnData>, context: any): MaterialTable<ColumnData>', gave the following error.
    Type '(newData: CreateRateData, oldData: ExistingRateData) => Promise<void>' is not assignable to type '(newData: ColumnData, oldData?: ColumnData | undefined) => Promise<any>'.
     84 |
     85 |   return (
  >  86 |     <MaterialTable
        |     ^^^^^^^^^^^^^^
  >  87 |       icons={tableIcons}
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  88 |       title="Stillingsprosent"
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  89 |       columns={columns}
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  90 |       data={filteredData}
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  91 |       isLoading={isLoading}
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  92 |       options={{ ...globalTableOptions }}
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  93 |       editable={{
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  94 |         onRowAdd: handleRowAdd,
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  95 |         onRowUpdate: handleRowUpdate,
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  96 |       }}
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  97 |       localization={norsk}
        | ^^^^^^^^^^^^^^^^^^^^^^^^
  >  98 |     />
        | ^^^^^^^
     99 |   );
    100 | }
```

Trulsaa commented:
> EmploymentRateTable

Så langt jeg forstår fra feilmeldingen så er Interfacet ColumnData du har laget egentlig RowData. Og både newData og oldData i funksjonene må da være av interfacet RowData.

Commit:
Made the compiler less angry 🦁
Improved some of the typing
- A small pinch of typescript Love
- Extended the RowData-interface
